### DISCUSSION TITLE
Google's official API key for search tasks (?)

### DISCUSSION THREAD

**Discussion by jakub-trzebiatowski** [myshell-ai/AIlice#40] [Q&A] [answered]
> However, for users who require frequent access to Google, I assume you'd be willing to endure the hassle of applying for a Google's official API key for search tasks. For these users, please open config.json and use the following configuration:

Sorry if it's obvious, but where/how do I apply for such a key? All I found is [Custom Search JSON API](https://developers.google.com/custom-search/v1/introduction), part of GCP, which is not a full Google Search if I understand correctly.

**Comment by stevenlu137** [comment, marked as answer]
I wrote too briefly in the document, yes, it's this API. If you're referring to the search scope being limited, that's actually not the case. When creating the API, you can choose whether to search the entire web. Additionally, although the free quota is 100 searches per day, you can purchase more search quotas. But my knowledge is limited to this. If you're referring to other limitations, please let me know, and we'll find ways to improve the search module.

**Reply by jakub-trzebiatowski** [reply]
I was confused by the official docs of [Programmable Search Engine](https://developers.google.com/custom-search), which state...

> Programmable Search Engine enables you to create a search engine for your website, your blog, or a collection of websites.

It's good to know "a collection of websites" can include "full Web"!
